### PR TITLE
Schema Design under New Notices docs #1196

### PR DESCRIPTION
**Pull request by Vidushi-GitHub**
Hi all, Please find the page/PR for schema design under new notices docs format. Attachments (#1205) and practices (#1206) would be nested under schema design later, once ready. @jracusin it's going parallel to producers, too much dropdown/childpages looked strange. As it's already under nested notices, it should be fine.

### CODE REVIEW

**Review by lpsinger — CHANGES_REQUESTED** [review]
Please hide the new docs behind the feature flag.

File: app/routes/docs.notices.producers.mdx [hunk 1]
```diff
@@ -44,6 +44,12 @@ also happy to work with the mission teams to help construct your alerts.
    your list of team members from Step 1 and your chosen Kafka topic prefix from Step 2.
       The GCN Team will reply after they have configured producer permissions for your team.
    </ProcessListItem>
+   <ProcessListItem>
+     <ProcessListHeading type="h3">
+       Draft Your Schema 
+     </ProcessListHeading>
+     As a GCN Notice producer, you have the flexibility to create your own instrument-specific schema. Please contribute your schema to our [Github repository](https://github.com/nasa-gcn/gcn-schema), placing it in a folder under <code>gcn/notices/<i>mission</i></code> and submit a pull request for the GCN Team to review. For details, please refer to the [schema design](/docs/producers/schema) documentation.
+   </ProcessListItem>
```
Comment (lpsinger) [inline app/routes/docs.notices.producers.mdx:52]: Hide this added section behind a feature flag.

File: app/routes/docs.notices.producers.mdx [hunk 2]
```diff
@@ -61,9 +67,16 @@ also happy to work with the mission teams to help construct your alerts.
...
       # any topic starting with 'mission.'
       topic = 'gcn.notices.mission.example'
-      data = b'...'  # any bytes
+
+      # JSON data converted to byte string format
+      data = json.dumps({
+         '$schema':'your/schema',
+         'key':'value'
+         ...
+      }).encode()  # any bytes
```
Comment (lpsinger) [inline app/routes/docs.notices.producers.mdx:79]:
```suggestion
      }).encode()
```

File: app/routes/docs.notices.producers.mdx [hunk 3]
```diff
@@ -61,9 +67,16 @@ also happy to work with the mission teams to help construct your alerts.
...
+
       # any topic starting with 'mission.'
       topic = 'gcn.notices.mission.example'
-      data = b'...'  # any bytes
+
+      # JSON data converted to byte string format
+      data = json.dumps({
+         '$schema':'your/schema',
+         'key':'value'
+         ...
```
Comment (lpsinger) [inline app/routes/docs.notices.producers.mdx:78]:
```suggestion
         '$schema': 'your/schema',
         'key': 'value'
         ...
```

- [ ] Add whitespace to conform to PEP 8.
- [ ] The value of `$schema` must be a URL.
- [ ] Hide this sample code behind a feature flag.

File: app/routes/docs.notices.schema.mdx [hunk 4]
```diff
@@ -0,0 +1,204 @@
+---
```
Comment (lpsinger) [inline app/routes/docs.notices.schema.mdx:1]: Hide this file behind a feature flag.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/docs.tsx [hunk 5]
```diff
@@ -17,6 +17,7 @@ export const handle = {
 
 export default function () {
   const enableSchemaBrowser = useFeature('SCHEMA')
+  const enableSchemaDoc = useFeature('NewDoc')
```
Comment (lpsinger) [inline app/routes/docs.tsx:20]: Use the `SCHEMA` feature, please.

File: app/routes/docs.notices.producersS.mdx [hunk 6]
```diff
@@ -0,0 +1,100 @@
+---
```
Comment (lpsinger) [inline app/routes/docs.notices.producersS.mdx:1]: Why is there this new file with the suffix S.mdx?

File: app/routes/docs.notices.producers.mdx [hunk 7]
```diff
@@ -61,6 +61,7 @@ also happy to work with the mission teams to help construct your alerts.
       # Connect as a producer.
       # Warning: don't share the client secret with others.
       producer = Producer(client_id='fill me in', client_secret='fill me in')
+
```
Comment (lpsinger) [inline app/routes/docs.notices.producers.mdx:64]: Undo unrelated whitespace change.

```suggestion
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/docs.notices.producersS.mdx (hunk 6 above)
Comment (Vidushi-GitHub) [reply app/routes/docs.notices.producersS.mdx:1]: I didn't find the way how to hide things new things in the file using feature flag, still working on it.

**Comment by Vidushi-GitHub** [comment]
@lpsinger we have added a new component to handle the conditional logic for what should be displayed in the sample code.
Is this ok or can be implemented in .mdx?

File: app/routes/docs.notices.schema.mdx [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+
+import { feature } from '~/lib/env.server'
+
+export function loader() {
+  return feature('SCHEMA') || new Response(null, { status: 404 })
+}
+
+# GCN Unified Schema
+
+The GCN Unified schema provides a standardized method for organizing GCN notices with common properties across different missions following the uniform unit system. It utilizes JSON (JavaScript Object Notation), a lightweight and easily machine-readable data-interchange format, to define the notice schema. The set of core schemas serves as the foundation for creating new GCN Notices alerts, also allows the inclusion of instrument-specific parameters.
```
Comment (lpsinger) [inline app/routes/docs.notices.schema.mdx:24]:
```suggestion
The GCN Unified Schema provides a standardized method for organizing GCN notices with common properties across different missions following the uniform unit system. It utilizes JSON (JavaScript Object Notation), a lightweight and easily machine-readable data-interchange format, to define the notice schema. The set of core schemas serves as the foundation for creating new GCN Notices alerts, also allows the inclusion of instrument-specific parameters.
```

Also mention JSON Schema.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/docs.notices.schema.mdx (hunk 8 above)
Comment (Vidushi-GitHub) [reply app/routes/docs.notices.schema.mdx:24]: Should re-name as "Unified JSON Schema"?

File: app/routes/docs.notices.unifiedschema.mdx [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+              "data_archive_page": "https://guano.swift.psu.edu/trigger_report?id=694215995",
+              "id": ["694215995"]
+            },
+            null,
+            2
+          )}
+        />
+      </Tab>
+    </Tabs>
+    Defining structures in JSON schema can get confusing, but we are happy to help.
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:189]: It's good to include an offer of help here, but can we do that without saying that the process is expected to be confusing?

File: app/routes/docs.notices.unifiedschema.mdx [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+
+    To validate your schemas and examples, execute the following command in the same location:
+
+    <Highlight language="sh" code="npm run validate" />
+
+  </ProcessListItem>
+  <ProcessListItem>
+    <ProcessListHeading type="h3">Submit Your Schema</ProcessListHeading>
+    
+    Once your schema and examples are validated, you can proceed to commit and push your local code to the [Git repository](https://github.com/nasa-gcn/gcn-schema).
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:205]: But they _can't_ push code to that repository. They can only push to a fork.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+
+    Change your working directory:
+    <Highlight language="sh" code="cd gcn-schema" />
+
+    Install the necessary packages:
+    <Highlight language="sh" code="npm i" />
+
+  </ProcessListItem>
+  <ProcessListItem>
+  <ProcessListHeading type="h3">Setting-Up Your Repository</ProcessListHeading>
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:47]:
```suggestion
  <ProcessListHeading type="h3">Setting Up Your Repository</ProcessListHeading>
```

**Review by lpsinger — CHANGES_REQUESTED** [review]
- [x] I question whether we should be so specific about how to open a pull request here. The process of opening a pull request is not specific to this project, and there are better walkthroughs elsewhere on the Web. It might be better to point to some other reference for that.
- [ ] If we do include some instructions about how to open a pull request, then those instructions here would need a lot of work. For one, these instructions don't include making  fork.
- [ ] I envisioned this document as being structured as a primer on our JSON Schema rather than as a set of step-by-step instructions. The "New Notice Producers" page is already in the form of step-by-step instructions, and now one of those steps points to a different set of instructions. I think that this will be confusing.

@dakota002, would you please help with the Git, GitHub, and general contributing issues here?

Perhaps we could put some general guidelines on contributing to https://github.com/nasa-gcn projects in the Contributing section.

File: app/components/ProducerSampleCode.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+          # Connect as a producer.
+          # Warning: don't share the client secret with others.
+          producer = Producer(client_id='fill me in', client_secret='fill me in')
+          # any topic starting with 'mission.'
+          topic = 'gcn.notices.mission'
+          # JSON data converted to byte string format
+          data = json.dumps({
+            '$schema': 'https://gcn.nasa.gov/schema/main/gcn/notices/mission/SchemaName.schema.json',
+            'key': 'value'
+          }).encode() # Parse URL of your schema
```
Comment (Vidushi-GitHub) [inline app/components/ProducerSampleCode.tsx:31]:
```suggestion
          }).encode()
```

[the remark this answers is not in the document]
Comment (Vidushi-GitHub) [reply app/components/ProducerSampleCode.tsx:31]: Yes, it's for JSON data conversion. I wrote to what user is supposed to change in their code.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+            null,
+            2
+          )}
+        />
+      </Tab>
+    </Tabs>
+</ProcessListItem>
+<ProcessListItem>
+  <ProcessListHeading type="h3">Defining Schema</ProcessListHeading>
+    We recommend exploring our [schema-browser](/docs/schema/stable/gcn) to get an idea of what common fields have already been defined in the [Core schema](https://github.com/nasa-gcn/gcn-schema/tree/main/gcn/notices/core), as well as the [json schema documentation](https://www.learnjsonschema.com/) to familiarize yourself with the various keywords and types. A good method to start formulating your structure is to think first of your data product as JSON fields. Consider the possible values that your pipelines will have for each field. In the following example, we have a schema definition for Swift/BAT-GUANO alerts. They have included the `unevaluatedProperties` property and set it to `false`, which our validator can use to verify that there are no extra properties that are undefined in the examples:
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:106]:
```suggestion
    We recommend exploring our [schema-browser](/docs/schema/stable/gcn) to get an idea of what common fields have already been defined in the [Core schema](https://github.com/nasa-gcn/gcn-schema/tree/v1.0.0/gcn/notices/core), as well as the [json schema documentation](https://www.learnjsonschema.com/) to familiarize yourself with the various keywords and types. A good method to start formulating your structure is to think first of your data product as JSON fields. Consider the possible values that your pipelines will have for each field. In the following example, we have a schema definition for Swift/BAT-GUANO alerts. They have included the `unevaluatedProperties` property and set it to `false`, which our validator can use to verify that there are no extra properties that are undefined in the examples:
```

File: app/routes/docs.notices.unifiedschema.mdx (hunk 9 above)
Comment (Vidushi-GitHub) [reply app/routes/docs.notices.unifiedschema.mdx:189]: I will re-word it!

File: app/routes/docs.notices.unifiedschema.mdx [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+    <Tabs>
+      <Tab label="Alert.schema.json">
+        <Highlight
+          language="json"
+          code={JSON.stringify(
+            {
+              "$id": "https://gcn.nasa.gov/schema/gcn/notices/swift/bat/guano/alert.schema.json",
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:113]:
```suggestion
              "$id": "https://gcn.nasa.gov/schema/main/gcn/notices/swift/bat/guano/alert.schema.json",
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+              "data_archive_page": "https://guano.swift.psu.edu/trigger_report?id=694215995",
+              "id": ["694215995"]
+            },
+            null,
+            2
+          )}
+        />
+      </Tab>
+    </Tabs>
+    Defining structures in JSON schema can get confusing, but we are happy to help.
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:189]:
```suggestion
While defining JSON schema structures, please don't hesitate to reach out with any questions about Schema field definitions or specific fields you are looking for.
```

File: app/routes/docs.notices.producers.mdx [hunk 16]
```diff
@@ -44,6 +47,14 @@ also happy to work with the mission teams to help construct your alerts.
    your list of team members from Step 1 and your chosen Kafka topic prefix from Step 2.
       The GCN Team will reply after they have configured producer permissions for your team.
    </ProcessListItem>
+   <WithFeature schema>
+      <ProcessListItem>
+         <ProcessListHeading type="h3">
+            Draft Your Schema 
+         </ProcessListHeading>
+         As a GCN Notice producer, you have the flexibility to create your own instrument-specific schema. Please contribute your schema to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema), placing it in a folder under <code>gcn/notices/<i>mission</i></code> and submit a pull request for the GCN Team to review. For details, please refer to the [schema design](/docs/producers/schema) documentation.
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.producers.mdx:55]:
```suggestion
         As a GCN Notice producer, you have the flexibility to create your own instrument-specific schema. Please contribute your schema to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema), placing it in a folder under <code>gcn/notices/<i>mission</i></code> and submit a pull request for the GCN Team to review. For details, please refer to the [schema design](/docs/producers/schema) documentation for step-by-step instructions.
```

**Comment by Vidushi-GitHub** [comment]
> * [x]  I question whether we should be so specific about how to open a pull request here. The process of opening a pull request is not specific to this project, and there are better walkthroughs elsewhere on the Web. It might be better to point to some other reference for that.
> * [ ]  If we do include some instructions about how to open a pull request, then those instructions here would need a lot of work. For one, these instructions don't include making  fork.
> * [ ]  I envisioned this document as being structured as a primer on our JSON Schema rather than as a set of step-by-step instructions. The "New Notice Producers" page is already in the form of step-by-step instructions, and now one of those steps points to a different set of instructions. I think that this will be confusing.
> 
> @dakota002, would you please help with the Git, GitHub, and general contributing issues here?
> 
> Perhaps we could put some general guidelines on contributing to https://github.com/nasa-gcn projects in the Contributing section.

[2] Shouldn't there be a section on pull request in contributing section, then that's useful anywhere in project.
[3] Ok, It looks good when written as step-by-step instructions and easy to follow for Creating Schema. Production is different than Designing Schema, I don't know how to link with New Producers, such that it's not confusing and looks clear. Maybe it's step before Production, then in producing, use your Schema.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+
+    To validate your schemas and examples, execute the following command in the same location:
+
+    <Highlight language="sh" code="npm run validate" />
+
+  </ProcessListItem>
+  <ProcessListItem>
+    <ProcessListHeading type="h3">Submit Your Schema</ProcessListHeading>
+    
+    Once your schema and examples are validated, you can proceed to commit and push your local code to your github repo.
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:186]:
```suggestion
    Once your schema and examples are validated, you can proceed to commit and push your local code to your GitHub repository.
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+
+    <Highlight language="sh" code="npm run validate" />
+
+  </ProcessListItem>
+  <ProcessListItem>
+    <ProcessListHeading type="h3">Submit Your Schema</ProcessListHeading>
+    
+    Once your schema and examples are validated, you can proceed to commit and push your local code to your GitHub repository.
+
+    Next, create a pull request to our [Git repository](https://github.com/nasa-gcn/gcn-schema). The GCN team will review and provide feedback as needed. Once you have incorporated the feedback, we will merge your pull request into main branch. Congratulations on successfully creating your mission schema! It is now ready for production and send alerts to GCN as described in [New Notice Producers](/docs/notices/producers).
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:188]:
```suggestion
    Next, create a pull request to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema). The GCN team will review and provide feedback as needed. Once you have incorporated the feedback, we will merge your pull request into main branch. Congratulations on successfully creating your mission schema! It is now ready for production and send alerts to GCN as described in [New Notice Producers](/docs/notices/producers).
```

**Comment by Vidushi-GitHub** [comment]
@dakota002 after merging changes from your repo, two checks are failed, please check if any of them is imp.

**Comment by dakota002** [comment]
@Vidushi-GitHub that is okay, it is just the code coverage ones. The build is still good

**Review by lpsinger — CHANGES_REQUESTED** [review]
I don't think a process list works for this. Most of the text looks pretty good. Please reformat as just sections with headings and paragraphs.

**Comment by lpsinger** [comment]
> @Vidushi-GitHub that is okay, it is just the code coverage ones. The build is still good

Yeah, I find that failed status pretty annoying. We don't have high enough coverage on this project for it to matter right now. See #1283 to turn that off.

**Comment by Vidushi-GitHub** [comment]
> I don't think a process list works for this. Most of the text looks pretty good. Please reformat as just sections with headings and paragraphs.

Okay, done. 
Though, I got habit of seeing those points, I felt those were kind of steps to follow to define the new Schema.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+
+## Fork the GitHub Repository
+
+To begin, initiate by forking your own copy of the [nasa-gcn/gcn-schema](https://github.com/nasa-gcn/gcn-schema) repository on GitHub. Next, proceed to clone a local version and follow the project configuration instructions available [instructions](https://gcn.nasa.gov/docs/contributing).
+
+## Setting Up Your Repository
+
+The repository's structure allows producers to create their own folders to define schemas on an instrument-specific level. The schema folder is named according to the format <code>gcn.notices.<i>mission</i></code>, where <i>mission</i> is your [Kafka Topic](/docs/notices/producers). Within it, you can define your own schema and example files. For example, schema definitions for IceCube alerts are located in the directory [`gcn/notices/icecube/`](https://github.com/nasa-gcn/gcn-schema/tree/main/gcn/notices/icecube).
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:32]: Please update these instructions to point to the new GitHub primer (/docs/contributing/github).

File: app/routes/docs.notices.unifiedschema.mdx [hunk 20]
```diff
@@ -0,0 +1,191 @@
...
+
+## Schema File Naming
+
+Naming convention for your new schema files should follow the pattern <code><i>SchemaName</i>.schema.json</code> for schema definitions.
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:36]:
```suggestion
Your new schema definition files should follow the naming convention <code><i>SchemaName</i>.schema.json</code>.
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 21]
```diff
@@ -0,0 +1,191 @@
...
+  </Tab>
+</Tabs>
+While defining JSON schema structures, please don't hesitate to reach out with any
+questions about Schema field definitions or specific fields you are looking for.
+
+## Validating Schema
+
+A validation pre-commit hook has been configured for this project. When you commit your updates, the validator will run to make sure that all of the schemas and their respective examples function correctly. This step also checks general formatting using the [Prettier](https://prettier.io/) formatter. To quickly format your code, run the following command in the root directory of your local repository. This will format spacing, line-breaks, etc. as needed.
+
+<Highlight language="sh" code="npx prettier -w ." />
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:181]: They should _never_ have to run this manually.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 22]
```diff
@@ -0,0 +1,191 @@
...
+            'https://guano.swift.psu.edu/trigger_report?id=694215995',
+          id: ['694215995'],
+        },
+        null,
+        2
+      )}
+    />
+  </Tab>
+</Tabs>
+While defining JSON schema structures, please don't hesitate to reach out with any
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:174]:
```suggestion

While defining JSON schema structures, please don't hesitate to reach out with any
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 23]
```diff
@@ -0,0 +1,191 @@
...
+questions about Schema field definitions or specific fields you are looking for.
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:175]: Reach out to whom? Add link to the contact page.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/docs.notices.unifiedschema.mdx [hunk 24]
```diff
@@ -0,0 +1,192 @@
...
+          id: ['694215995'],
+        },
+        null,
+        2
+      )}
+    />
+  </Tab>
+</Tabs>
+
+While defining JSON schema structures, please don't hesitate to reach out with any
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:175]:
```suggestion
While defining JSON schema structures, please don't hesitate to [reach out](https://gcn.nasa.gov/contact) with any
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 25]
```diff
@@ -0,0 +1,192 @@
...
+</Tabs>
+
+While defining JSON schema structures, please don't hesitate to [reach out](https://gcn.nasa.gov/contact) with any
+questions about Schema field definitions or specific fields you are looking for.
+
+## Validating Schema
+
+A validation pre-commit hook has been configured for this project. When you commit your updates, the validator will run to make sure that all of the schemas and their respective examples function correctly. This step also checks general formatting using the [Prettier](https://prettier.io/) formatter. To quickly format your code, run the following command in the root directory of your local repository. This will format spacing, line-breaks, etc. as needed.
+
+<Highlight language="sh" code="npx prettier -w ." />
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:182]:
```suggestion
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 26]
```diff
@@ -0,0 +1,191 @@
...
+    />
+  </Tab>
+</Tabs>
+
+While defining JSON schema structures, please don't hesitate to [reach out](https://gcn.nasa.gov/contact) with any
+questions about Schema field definitions or specific fields you are looking for.
+
+## Validating Schema
+
+A validation pre-commit hook has been configured for this project. When you commit your updates, the validator will run to make sure that all of the schemas and their respective examples function correctly. This step also checks general formatting using the [Prettier](https://prettier.io/) formatter. To quickly format your code, run the following command in the root directory of your local repository. This will format spacing, line-breaks, etc. as needed.
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:180]:
```suggestion
The Git repository is configured to automatically format and validate your code at the time that you commit it. This means when you commit your updates, the validator will run to make sure that all of the schemas and their respective examples function correctly. This step also checks general formatting using the [Prettier](https://prettier.io/) formatter. To quickly format your code, run the following command in the root directory of your local repository. This will format spacing, line-breaks, etc. as needed.
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 27]
```diff
@@ -0,0 +1,191 @@
...
+
+To validate your schemas and examples, execute the following command in the same location:
+
+<Highlight language="sh" code="npm run validate" />
+
+## Submit Your Schema
+
+Once your schema and examples are validated, you can proceed to commit and push your local code to your GitHub repository.
+
+Next, create a pull request to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema). The GCN team will review and provide feedback as needed. Once you have incorporated the feedback, we will merge your pull request into main branch. Congratulations on successfully creating your mission schema! It is now ready for production and send alerts to GCN as described in [New Notice Producers](/docs/notices/producers).
```
Comment (Vidushi-GitHub) [inline app/routes/docs.notices.unifiedschema.mdx:191]:
```suggestion
Next, create a pull request to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema), for details see [GitHub primer](/docs/contributing/github). The GCN team will review and provide feedback as needed. Once you have incorporated the feedback, we will merge your pull request into main branch. Congratulations on successfully creating your mission schema! It is now ready for production and send alerts to GCN as described in [New Notice Producers](/docs/notices/producers).
```

File: app/routes/docs.notices.producers/route.mdx [hunk 28]
```diff
@@ -44,6 +48,14 @@ also happy to work with the mission teams to help construct your alerts.
    your list of team members from Step 1 and your chosen Kafka topic prefix from Step 2.
       The GCN Team will reply after they have configured producer permissions for your team.
    </ProcessListItem>
+   <WithFeature schema>
+      <ProcessListItem>
+         <ProcessListHeading type="h3">
+            Draft Your Schema 
+         </ProcessListHeading>
+         As a GCN Notice producer, you can create your own instrument-specific schema. Please contribute your schema to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema), placing it in a folder under <code>gcn/notices/<i>mission</i></code> and submit a pull request for the GCN Team to review. For details, please refer to the [schema design](/docs/producers/schema) documentation.
```
Comment (lpsinger) [inline app/routes/docs.notices.producers/route.mdx:56]:
```suggestion
         As a GCN Notice producer, you can create your own instrument-specific schema. Please contribute your schema to our [GitHub repository](https://github.com/nasa-gcn/gcn-schema), placing it in a folder under <code>gcn/notices/<i>mission</i></code> and submit a pull request for the GCN Team to review. For details, please refer to the [schema documentation](/docs/producers/schema).
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 29]
```diff
@@ -0,0 +1,191 @@
...
+        null,
+        2
+      )}
+    />
+  </Tab>
+</Tabs>
+
+## Defining Schema
+
+We recommend exploring our [schema-browser](/docs/schema/stable/gcn) to get an idea of what common fields have already been defined in the [Core schema](https://github.com/nasa-gcn/gcn-schema/tree/main/gcn/notices/core), as well as the [json schema documentation](https://www.learnjsonschema.com/) to familiarize yourself with the various keywords and types. A good method to start formulating your structure is to think first of your data product as JSON fields. Consider the possible values that your pipelines will have for each field. In the following example, we have a schema definition for Swift/BAT-GUANO alerts. They have included the `unevaluatedProperties` property and set it to `false`, which our validator can use to verify that there are no extra properties that are undefined in the examples:
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:85]:
```suggestion
We recommend exploring our [schema browser](/docs/schema/stable/gcn) to get an idea of what common fields have already been defined in the [Core schema](https://github.com/nasa-gcn/gcn-schema/tree/main/gcn/notices/core), as well as the [JSON Schema documentation](https://json-schema.org/) to familiarize yourself with the various keywords and types. A good method to start formulating your structure is to think first of your data product as JSON fields. Consider the possible values that your pipelines will have for each field. In the following example, we have a schema definition for Swift/BAT-GUANO alerts. They have included the `unevaluatedProperties` property and set it to `false`, which our validator can use to verify that there are no extra properties that are undefined in the examples:
```

Also, break this long paragraph into 2-3 shorter ones.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 30]
```diff
@@ -0,0 +1,191 @@
+---
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:1]: Rename this file to `docs.notices.schema.mdx`.

File: app/routes/docs.notices.unifiedschema.mdx [hunk 31]
```diff
@@ -0,0 +1,191 @@
...
+
+import { feature } from '~/lib/env.server'
+
+export function loader() {
+  return feature('SCHEMA') || new Response(null, { status: 404 })
+}
+
+# GCN Unified Schema
+
+The GCN Unified Schema provides a standardized method for organizing GCN notices with common properties across different missions following the uniform unit system. It utilizes JSON (JavaScript Object Notation), a lightweight and easily machine-readable data-interchange format, to define the notice schema. The set of core schemas serves as the foundation for creating new GCN Notices alerts, also allows the inclusion of instrument-specific parameters.
```
[the remark this answers is not in the document]
Comment (lpsinger) [reply app/routes/docs.notices.unifiedschema.mdx:24]: I don't think you've addressed this comment yet, have you? Namely, explaining the relationship between JSON (the schemaless data serialization) and JSON Schema (the standard for declaring the allowed fields and types in a JSON document).

File: app/routes/docs.notices.unifiedschema.mdx [hunk 32]
```diff
@@ -0,0 +1,191 @@
...
+
+## Schema File Naming
+
+Your new schema definition files should follow the naming convention <code><i>SchemaName</i>.schema.json</code>.
+Examples corresponding to these schemas should follow the format, <code><i>SchemaName[.OptionalSpecification]</i>.example.json</code>. Both the <i>SchemaName</i> and <i>OptionalSpecification</i> example should be written in CamelCase formatting (ex., YourSchemaName, thisIsAlsoValid, etc.). This naming convention allows our [schema-browser](/docs/schema/stable/gcn) to generate pages based directly on the files in the latest tag.
+
+## Sample Code
+
+After setting up your local version, you can use the provided code samples to get started, which already include commonly used keys. Note that the `$schema` field in the example has the same value as the `$id` field in the schema.
+This ensures our validator script to identify which schema examples to compare. Make sure to update these files to reflect the mission and schema names. Then, you can update the `title` and `definition` fields to provide context for other users who will use this schema.
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:42]:
```suggestion
This allows our validator script to identify which schema examples to compare. Make sure to update these files to reflect the mission and schema names. Then, you can update the `title` and `definition` fields to provide context for other users who will use this schema.
```

File: app/routes/docs.notices.unifiedschema.mdx [hunk 33]
```diff
@@ -0,0 +1,191 @@
...
+    />
+  </Tab>
+</Tabs>
+
+While defining JSON schema structures, please don't hesitate to [reach out](https://gcn.nasa.gov/contact) with any
+questions about Schema field definitions or specific fields you are looking for.
+
+## Validating Schema
+
+The Git repository is configured to automatically format and validate your code at the time that you commit it. This means when you commit your updates, the validator will run to make sure that all of the schemas and their respective examples function correctly. This step also checks general formatting using the [Prettier](https://prettier.io/) formatter. To quickly format your code, run the following command in the root directory of your local repository. This will format spacing, line-breaks, etc. as needed.
```
Comment (lpsinger) [inline app/routes/docs.notices.unifiedschema.mdx:180]: "To quickly format your code, run the following command in the root directory of your local repository." What command? Or do you want to just remove this sentence?

**Review by lpsinger — CHANGES_REQUESTED** [review]
I think that this is getting close...

**Review by lpsinger — APPROVED** [review]
The wording still needs some work, but I'll touch it up on the main branch.
